### DISCUSSION TITLE
Custom connector: reading account-specific tokens from DB?

### DISCUSSION THREAD

**Discussion by pai911** [airbytehq/airbyte#28348] [Connector Questions] [answered]
Hi there!

I’m developing a connector with CDK to get data from an API.

The connector is designed so that we can set up each Source with a different accountID; for an accountID, multiple user tokens may be available, and they are stored in a database.

My question is: is it okay for my connector to read tokens from my database, and rotate them in each sync?

Will this be considered a bad practice? or is it something expected when developing a connector?

**Comment by marcosmarxm** [comment, marked as answer]
The best path probably would be to have a API endpoint in your application to get the tokens, if isn't possible probably you can customize CDK functions to hit the database and get the new tokens.

**Reply by pai911** [reply]
thank you for the insights! 

so that way, we can leverage the sub-stream feed the tokens from our API endpoint to the sub-streams (other API endpoints).
